include/zephyr/drivers/charger.h

```diff
@@ -272,6 +281,24 @@ static inline int z_impl_charger_set_prop(const struct device *dev, const charge
 	return api->set_property(dev, prop, val);
 }
 
+/**
+ * @brief Enable or disable a charge cycle
+ *
+ * @param dev Pointer to the battery charger device
+ * @param enable true enables a charge cycle, false disables a charge cycle
+ *
+ * @retval 0 if successful
+ * @retval < 0 if setting property failed
```

Comment:
`@retval` should be used to enumerate all the possible actual return values, so "< 0" is not one of them (also, the white space between `<` and `0` is problematic anyways as can bn seen [here](https://builds.zephyrproject.io/zephyr/pr/66128/docs/doxygen/html/group__charger__interface.html#gace1ea9841574d75d314db078df5a0b3a))

Should be 
```suggestion
 * @return 0 if successful, or a negative errnor if setting property failed
```

or stick to `@retval` if it's possible to enumerate all the possible error codes.

Reply:
I have seen some chargers with a global fault bit. If this bit is set it indicates "don't bother trying to initiate a charge cycle because there is some kind of lingering fault condition".

I could imagine this handler checking this condition on chargers that have this ability. In the case where a global fault is lingering I think `-EINVAL` would be an appropriate response. Let me know if you agree, and I will explicitly enumerate both as `@retval` values.

Reply:
ack, done